File: src/jvmception/instructions/InstructionConst.java
package jvmception.instructions;

import jvmception.CallFrame;
import jvmception.jvmtypes.IUnitSerializable;

public class InstructionConst extends Instruction {

	private final IUnitSerializable constValue;
	
	public InstructionConst(IUnitSerializable constValue) {
		this.constValue = constValue;
	}
	
	@Override
	public void execute(CallFrame frame) {
		frame.push(constValue);
	}

}

File: src/jvmception/instructions/InstructionConst.cs
public class InstructionConst : Instruction
{

    private readonly IUnitSerializable constValue;

    public InstructionConst(IUnitSerializable constValue)
    {
        this.constValue = constValue;
    }

    public override void execute(CallFrame frame)
    {
        frame.push(constValue);
    }
}
